services/webhook/elder_eval/harvest_review_findings.py

```
@@ -0,0 +1,452 @@
+    can't be fairly scored as a miss.
+    """
+    elder_index: dict[tuple[str, int, str], list[int | None]] = {}
+    for row in grug_findings:
+        try:
+            key = (str(row["repo"]), int(row["pr"]), str(row.get("path", "")))
+        except (KeyError, TypeError, ValueError):
+            continue
+        line = row.get("line")
+        elder_index.setdefault(key, []).append(int(line) if line is not None else None)
+
+    def elder_has(repo: str, pr: int, path: str, line: Any) -> bool:
```

Ruff (0.15.20)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `line`

(ANN401)